azurerm/internal/services/network/resource_arm_virtual_hub_connection.go

```diff
@@ -0,0 +1,301 @@
+
+	d.Set("name", id.Name)
+	d.Set("virtual_hub_id", virtualHub.ID)
+	d.Set("hub_to_vitual_network_traffic_allowed", props.AllowHubToRemoteVnetTransit)
+	d.Set("vitual_network_to_hub_gateways_traffic_allowed", props.AllowRemoteVnetToUseHubVnetGateways)
+	d.Set("internet_security_enabled", props.EnableInternetSecurity)
+
+	remoteVirtualNetworkId := ""
+	if props.RemoteVirtualNetwork != nil && props.RemoteVirtualNetwork.ID != nil {
+		remoteVirtualNetworkId = *props.RemoteVirtualNetwork.ID
+	}
+	d.Set("remote_virtual_network_id", remoteVirtualNetworkId)
+
+	return nil
+}
+
+func resourceArmVirtualHubConnectionDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Network.VirtualHubClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.ParseVirtualHubConnectionID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resourceGroup := id.ResourceGroup
+	virtualHubName := id.VirtualHubName
+
+	locks.ByName(virtualHubName, virtualHubResourceName)
+	defer locks.UnlockByName(virtualHubName, virtualHubResourceName)
+
+	virtualHub, err := client.Get(ctx, resourceGroup, virtualHubName)
+	if err != nil {
+		if utils.ResponseWasNotFound(virtualHub.Response) {
+			return fmt.Errorf("Virtual Hub %q was not found in Resource Group %q", virtualHubName, resourceGroup)
+		}
+
+		return fmt.Errorf("Error retrieving Virtual Hub %q (Resource Group %q): %+v", virtualHubName, resourceGroup, err)
+	}
+
+	if virtualHub.VirtualHubProperties == nil {
+		return fmt.Errorf("Error retrieving Virtual Hub %q (Resource Group %q): `properties` was nil", virtualHubName, resourceGroup)
+	}
+
+	if virtualHub.VirtualHubProperties.VirtualNetworkConnections == nil {
+		return fmt.Errorf("Error retrieving Virtual Hub %q (Resource Group %q): `properties.VirtualNetworkConnections` was nil", virtualHubName, resourceGroup)
+	}
+
+	var newConnections []network.HubVirtualNetworkConnection
+	for _, connection := range *virtualHub.VirtualHubProperties.VirtualNetworkConnections {
+		if connection.Name == nil {
+			continue
+		}
+
+		if *connection.Name == id.Name {
+			continue
+		}
+
+		newConnections = append(newConnections, connection)
+	}
+	virtualHub.VirtualHubProperties.VirtualNetworkConnections = &newConnections
+
+	future, err := client.CreateOrUpdate(ctx, resourceGroup, virtualHubName, virtualHub)
+	if err != nil {
+		return fmt.Errorf("Error removing Connection %q to Virtual Hub %q (Resource Group %q): %+v", id.Name, virtualHubName, resourceGroup, err)
+	}
+
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for removal of Connection %q to Virtual Hub %q (Resource Group %q): %+v", id.Name, virtualHubName, resourceGroup, err)
+	}
+
+	return nil
+}
+
+func findVirtualHubConnection(name string, connections []network.HubVirtualNetworkConnection) (conn *network.HubVirtualNetworkConnection, index int) {
```

Comment:
I didn't notice the index used anywhere
```suggestion
func findVirtualHubConnection(name string, connections []network.HubVirtualNetworkConnection) *network.HubVirtualNetworkConnection {
```

Reply:
Updated